index.js

```diff
@@ -190,6 +190,11 @@ module.exports = class Hypercore extends EventEmitter {
     })
 
     s._passCapabilities(this)
+
+    if (opts.encryptionKey) {
```

Comment:
&& !this.encryption mb?

Reply:
That depends on whether or not there's a use case for this I guess:

https://github.com/hypercore-protocol/hypercore-next/blob/c83657df93f7d9983c4434bc45f4351ad999c93c/test/encryption.js#L138-L149

Reply:
Fair. Does it make sense to check if the same key is being used on the existing encryption object?

Reply:
To avoid making a new one (unsure of the cost)